### PR TITLE
Refactor testrun retry

### PR DESCRIPTION
**Pull request by OlegLoewen**
**What this PR does / why we need it**:
Refactor testrun retry

**Release note**:

```
Refactor testrun retry
```

### CODE REVIEW

File: pkg/testrunner/types.go [hunk 1]
```diff
@@ -93,8 +93,8 @@ type Metadata struct {
 	// Additional annotations form the testrun or steps
 	Annotations map[string]string `json:"annotations,omitempty"`
 
-	// Tells whether testrun failed first but succeeded after a retry
-	Flaked int `json:"flaked,omitempty"`
+	// Represents how many retries the testrun had
+	Retries int `json:"flaked,omitempty"`
```
Comment (dguendisch) [inline pkg/testrunner/types.go]:
```suggestion
	Retries int `json:"retries,omitempty"`
```

Comment (OlegLoewen) [reply pkg/testrunner/types.go]: thx, done

File: pkg/testrunner/testrunner.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ package testrunner
 import (
 	"fmt"
 	"github.com/go-logr/logr"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (schrodit) [inline pkg/testrunner/testrunner.go]: nit metav1

File: pkg/testrunner/testrunner.go [hunk 3]
```diff
@@ -68,18 +69,17 @@ func (rl RunList) Run(log logr.Logger, tmClient kubernetes.Interface, namespace,
...
-					if flakeAttempt != 0 {
-						rl[i].Metadata.Flaked++
-					}
 					// testrun was successful, break retry loop
 					break
 				}
-				if flakeAttempt != maxFlakeAttempts {
+				if attempt < maxFlakeAttempts {
 					// clean status and name of testrun if it's failed to ignore it, since a retry will be initiated
+					log.V(3).Info(fmt.Sprintf("testrun failed, retry %d/%d. testrun", attempt, maxFlakeAttempts))
```
Comment (schrodit) [inline pkg/testrunner/testrunner.go]: I think we should make it a more important message
```suggestion
					log.Info(fmt.Sprintf("testrun failed, retry %d/%d. testrun", attempt, maxFlakeAttempts))
```

Comment (OlegLoewen) [reply pkg/testrunner/testrunner.go]: done.

File: pkg/testrunner/testrunner.go (hunk 2 above)
Comment (OlegLoewen) [reply pkg/testrunner/testrunner.go]: done.

**Pull request merged**